conans/test/functional/configuration/proxies_conf_test.py

```diff
@@ -21,21 +21,46 @@ def test_requester(self):
 [proxies]
 https=None
 no_proxy=http://someurl,http://otherurl.com
-http=http:/conan.url
+http=http://conan.url
         """
         save(client.cache.conan_conf_path, conf)
         client.cache.invalidate()
         requester = get_basic_requester(client.cache)
 
         def verify_proxies(url, **kwargs):
-            self.assertEqual(kwargs["proxies"], {"https": None, "http": "http:/conan.url"})
+            self.assertEqual(kwargs["proxies"], {"https": None, "http": "http://conan.url"})
             return "mocked ok!"
 
         requester._requester.get = verify_proxies
         self.assertEqual(os.environ["NO_PROXY"], "http://someurl,http://otherurl.com")
 
         self.assertEqual(requester.get("MyUrl"), "mocked ok!")
 
+    def test_requester_with_host_specific_proxies(self):
+        client = TestClient()
+        conf = """
+[proxies]
+https=http://conan.url
+  only.for.this.conan.url = http://special.url
```

Comment:
I think we would need a test with more than 1 host line.

Reply:
Added as requested